src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -129,14 +129,27 @@ protected Object readResolve() {
             c.cloud = this;
             // Apply a label that associates an instance configuration with
             // this cloud provider
-            c.appendLabel(CLOUD_ID_LABEL_KEY, String.valueOf(this.name.hashCode()));
+            c.appendLabel(CLOUD_ID_LABEL_KEY, getInstanceUniqueId());
 
             // Apply a label that identifies the name of this instance configuration
             c.appendLabel(CONFIG_LABEL_KEY, c.namePrefix);
         }
         return this;
     }
 
+    public String getInstanceUniqueId() {
+        // Semi unique ID
+        return String.valueOf(name.hashCode());
+    }
+    
+    public ComputeClient getClient() {
```

Comment:
Want me to create javadocs for that plain getters too?
I dont think its worth..
That one is not that simple :)

Reply:
I think at least a "@returns" annotation is helpful for javadoc readers, but I won't hold up for the PR for it :)